llama.cpp

```diff
@@ -8397,10 +8397,80 @@ void llama_sample_temp(struct llama_context * ctx, llama_token_data_array * cand
     }
 }
 
+
 void llama_sample_temperature(struct llama_context * ctx, llama_token_data_array * candidates_p, float temp) {
     llama_sample_temp(ctx, candidates_p, temp);
 }
 
+void llama_sample_entropy(struct llama_context * ctx, llama_token_data_array * candidates_p, float temp, float min_temp = 0, float max_temp = 2.0f) {
```

Comment:
It'd be good to hide all of the debugging printfs in this function behind the debugmode flag.